src/domain/request/Request.js

```diff
@@ -11,10 +11,6 @@ class Request {
     this.state = state ? new RequestState(state) : RequestStateFactory.createState(validFrom);
   }
 
-  getRawUrl() {
```

Comment:
We strive for rich domain objects that know their own capabilities. `RequestUrlParser` was always meant more as an internal utility for `Request` (in Java we'd give it package scope). Maybe there's some other way? Polymorphism comes to mind but I'm sure there are other possibilities. We could rethink the interface of a `Request` domain object to better fit our use cases (here specifically `FetchDataUseCase`).

Reply:
There are some distinct features (such as getRawUrl) that are going to be only used by some fetchers. I take the point about polymorphism - Request could be an abstraction and we could implement RandomRequest or RestRequest. This might however be redundant if there would be 1-to-1 relation between Request and Fetcher implementation. I think we should revisit this after implementing Random Provider, but for now let's leave it as is.

Reply:
Maybe we should have different use cases and some kind of strategy to choose from them? I'm reluctant to accept it the way it is now. By "leaving it as is" you mean to not merge this for now?

Reply:
> Maybe we should have different use cases and some kind of strategy to choose from them? 
I agree. This PR is a first step for achieving this.

> By "leaving it as is" you mean to not merge this for now?
Originally I wanted to merge this asap, then merge random provider once it's ready and then write something like a "strategy picker" refactor you mentioned before.
But if you would like me to, I could do it the hard way and design "strategy" picker before merging random provider :) .

Reply:
I think there's some middle ground between merging this as is and producing a large PR with both the provider and the strategy.

How about we create a `RequestExecutorFactory` which would create (select?) a use case based on the request. Each executor would return a response given a request. We'd have different request executors:

1. One that basically does what `fetchAndSelectData` function from `ExecuteReadyRequestsUseCase` does - maybe `UrlRequestExecutor`
1. One that can provide a random value based `RandomValueRequestExecutor`

This way we can keep the domain pure and segregate the concerns. `UrlRequestExecutor` would handle all the URL-related stuff so that the `Request` does not need to know about it at all.